### DISCUSSION TITLE
Communication lost after a while

### DISCUSSION THREAD

**Discussion by tafx** [kellerza/sunsynk#320] [Q&A]
Hi, I would like to ask for a help.
My setup: 2pcs 12 kW Deye 3ph LV inverters in parallel, master connected via RS485 "Modbus" port to waveshare RS485 to ETH(B) converter, addon "dev".

Communication is working only limited time period. Usually 3 to 30 mins. After that I got timeout errors:
"09:50:17 ERROR   OSError in callback read_deye_1: timeout reading 1 registers from 141; timeout reading 1 registers from 146; timeout reading 5 registers from 587; timeout reading 1 registers from 609; timeout reading 7 registers from 619; timeout reading 1 registers from 636; timeout reading 4 registers from 650; timeout reading 8 registers from 672 poll_need_to_read"

Addon restart helps to establish communication. Waveshare converter restart did not help.

Do you have any idea how to solve that?
Thanks.

**Comment by dimmon-ua** [comment]
The same problem 12 kW Deye 3ph LV after a while:

`11:45:08 ERROR   OSError in callback read_ss: timeout reading 6 registers from 141; timeout reading 1 registers from 500; timeout reading 1 registers from 552 poll_need_to_read

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/timer_callback.py", line 80, in wrap_callback
    await cb_call

  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/sensor_callback.py", line 109, in callback_sensor
    await ist.read_sensors(

  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/a_inverter.py", line 53, in read_sensors
    await self.inv.read_sensors(sensors)

  File "/usr/local/lib/python3.11/site-packages/sunsynk/sunsynk.py", line 133, in read_sensors
    raise IOError("; ".join(errs))

OSError: timeout reading 6 registers from 141; timeout reading 1 registers from 500; timeout reading 1 registers from 552 poll_need_to_read`
